### ISSUE TITLE
Incompatible override of overloaded function not caught

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#3189]
On current Pylance/Pyright, I get no errors on this file:

```python
from typing import Generic, TypeVar, Union, overload, Any

VT = TypeVar("VT")
T = TypeVar("T")


class X(Generic[VT]):
    @overload
    def pop(self) -> VT:
        ...

    @overload
    def pop(self, __default: Union[VT, T]) -> Union[VT, T]:
        ...

    def pop(self, __default: Any = None) -> Any:
        ...


class Child(X[VT]):
    def pop(self, __default: VT) -> VT:
        ...


class Child2(X[VT]):
    @overload
    def pop(self) -> VT:
        ...

    @overload
    def pop(self, __default: VT) -> VT:
        ...

    def pop(self, __default: Any = None) -> Any:
        ...
```

But both `Child` and `Child2` override `pop` incompatibly. (Found this in https://github.com/jab/bidict/pull/242#discussion_r825381187 )

**Comment by erictraut** [comment]
Pyright intentionally doesn't do any override validation when overloads are involved. There's really no good way (as far as I know) to determine whether one set of overloads is compatible with another. So rather than risk false positive errors, pyright doesn't attempt to detect override compatibilities in this case.

**Comment by JelleZijlstra** [comment]
Interesting, I hadn't realized that. I don't know what mypy does internally, but in pyanalyze for checking the compatibility of two overloaded signatures L and R, I check that R is compatible with all of the component signatures in L.
